### ISSUE TITLE
Question: Is the ConsoleLogger safe to use in production?

### ISSUE THREAD

**Issue by andrewlock** [aspnet/Logging#695]
I was configuring the logging for a new ASP.NET Core app, and I had the vague memory that the `ConsoleLogger` was not supposed to be used in production. 

Searching around, all I could find was a couple of comments on [this issue](https://github.com/aspnet/IISIntegration/issues/245#issuecomment-238666662) that suggested that a global lock in the `ConsoleLogger` library could cause performance issues. 

Comparing the `ConsoleLogger`  code between 1.0 and 2.0, it looks like a previous `lock()` usage has been removed. 

Does that mean the `ConsoleLogger` is now safe to use in production, or were these comments referring to a different global lock?

Thanks

**Comment by BrennanConroy** [comment]
I don't think it has ever been unsafe. It's just you will probably take a performance hit if you enable it.

**Comment by andrewlock** [comment]
Sure, that's what I meant by unsafe though - you were advised not to use it in production due to the performance impact (as mentioned a couple of times in [this issue](https://github.com/aspnet/IISIntegration/issues/245#issuecomment-238666662)). Are those performance concerns still there, or have they been alleviated by the rewrite to remove the lock?

**Comment by BrennanConroy** [comment]
Perf has been improved and now the logging happens on a background thread so individual calls to log should be fairly fast. They do still access a lock, but it is just to add the log to a queue for later processing.

I haven't seen any perf numbers of before and after though so I can't recommend for or against it.

**Comment by nphmuller** [comment]
Same question I have. The performance warning has been removed from the docs for [2.x](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging?tabs=aspnetcore2x#console),
 bit is still there for [1.x](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging?tabs=aspnetcore1x#console). So my guess is that if there's still an impact it would be negligible.

Would be great to have a definite answer though.

**Comment by BrennanConroy** [comment]
@glennc

**Comment by andrewlock** [comment]
Having looked through the code, the answer given here, (and I _vaguely_ remember hearing something mentioned on the community standup) I'm pretty sure this is safe for production now.

**Comment by pakrym** [comment]
@andrewlock console logger still uses `BlockingCollection` with size limit of 1024 so Log calls would start blocking when it gets reached. So very detailed console logs should be enabled with caution.

**Comment by andrewlock** [comment]
Thanks for the update @pakrym, that's good to know. Definitely something to bear in mind, especially as logging to the console is the preferred logging mechanism in e.g. Docker. 

I wonder if there should still be a mention of this somewhere in the docs, given that it could be difficult to diagnose slow requests, if you're not aware of this potential bottleneck.
